www/common/drive-ui.js

```diff
@@ -4551,178 +4590,211 @@ define([
-                $element.appendTo($list);
-                $element.find('>.cp-app-drive-element-row').contextmenu(openContextMenu('tree'));
-                if (isSharedFolder) {
-                    $element.find('>.cp-app-drive-element-row')
-                        .addClass('cp-app-drive-element-sharedf');
+                if (subfolder) {
+                    var subRoot = isSharedFolder ? 
+                        Util.find(manager, ['folders', sfId, 'proxy', manager.user.userObject.ROOT]) :
+                        root[key];
+                    if (subRoot) {
+                        data.content[randomKey].content = buildTreeData(subRoot, pathForRecursion, currentPath).content;
+                    }
+                }
+            }); 
+            return data;
+        };
+
+        var createTree = function ($container, path) {
+            var root = manager.find(path);
+
+            var isRoot = manager.comparePath([ROOT], path);
+            var rootName = ROOT_NAME;
+            if (APP.newSharedFolder && priv.isEmbed && isRoot) {
+                var newSFPaths = manager.findFile(Number(APP.newSharedFolder));
+                if (newSFPaths.length) {
+                    path = newSFPaths[0];
+                    path.push(ROOT);
+                    root = manager.find(path);
+                    rootName = manager.getSharedFolderData(APP.newSharedFolder).title;
                 }
-                if (sfId && !editable) {
-                    $element.attr('data-ro', true);
+            }
+            
+            // don't try to display what doesn't exist
+            if (!root) { return; }
+            
+            // build tree data structure
+            var treeData = buildTreeData(root, path, currentPath);
+            var data = {
+                root: {
+                    name: rootName,
+                    icon: $(Icons.get('drive')),
+                    content: treeData.content
                 }
-                if (!subfolder) { return; }
-                createTree($element, newPath);
+            };
+            var openFolders = LS.getOpenedFolders();
+            var filteredOpenFolders = openFolders.filter(function(p) {
+                return p && p.length > 0 && p[0] === path[0];
             });
+            var treeElement = UIElements.getTree(data, {
+                currentPath: currentPath,
+                rootPath: path,
+                events: {
+                    onFolderClicked: function (clickedPath, event) {
+                        // Check for shared folder restrictions
+                        var pathToCheck = clickedPath.slice(1); // Remove 'root'
+                        if (pathToCheck.length > 0) {
+                            var checkRoot = manager.find([ROOT]);
+                            var firstKey = pathToCheck[0];
+                            if (checkRoot && checkRoot[firstKey]) {
```

Comment:
It seems that this aprt will only check for "deleted" and "restricted" shared folders if they are directly in the drive root. Shared folders can be located inside other classic folders and we should also apply the check to them.
If you want to get the shared folder id, you can use something like:
```js
var sfId = manager.isInSharedFolder(clickedPath);
```